taichi/runtime/program_impls/vulkan/vulkan_program.cpp

```diff
@@ -222,6 +209,11 @@ std::unique_ptr<KernelLauncher> VulkanProgramImpl::make_kernel_launcher() {
   return std::make_unique<gfx::KernelLauncher>(std::move(cfg));
 }
 
+DeviceCapabilityConfig VulkanProgramImpl::get_device_caps() {
```

Comment:
I think we may need a `GfxProgramImpl` base class for these functions shared in all the gfx backends in the future.

Reply:
Agree!